dotnet.exe nugetize My.sln errors if one project doesn't PackageReference Nugetizer

## Describe the Bug

dotnet.exe nugetize My.sln errors if one project doesn't PackageReference Nugetizer

## Steps to Reproduce

Create a solution with two projects, /A/A.csproj and /B/B.csproj and only A.csproj references Nugetizer:

```xml
<PackageReference Include="Nugetizer" Version="*" />
```

run `dotnet.exe nugetize` on the top-level folder containing the solution file.

## Expected Behavior

Plain Old CSProj should not need to know about nugetizer.

Alternatively, or additionally, there should be a nugetize option to warn or silence warnings related to projects not having a PackageReference to Nugetizer.

Alternatively, it seems like there is some additional error checking that should happen on the csproj itself, such that it could check that PackageReference Include="Nugetizer" is present or not.  However, that could have versioning issues depending on how dotnet-nugetize is built due to assembly version mismatches between Nugetizer and dotnet-nugetize.

## Exception with Stack Trace

<!-- If you see an exception, put the WHOLE THING here. -->

```text
PS D:\Users\John.Zabroski\source\infrastructure2\master\Prototype> dotnet.exe nugetize
D:\Users\John.Zabroski\source\infrastructure2\master\Prototype\B.csproj : error MSB4057: The target "GetPackageContents" does not exist in the project.
```

## Version Info

0.7

## Additional Info

n/a

--- a/src/dotnet-nugetize/EmbeddedResource.cs
+++ b/src/dotnet-nugetize/EmbeddedResource.cs
@@ -1,0 +1,37 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+static class EmbeddedResource
+{
+    static readonly string baseDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+
+    public static string GetContent(string relativePath)
+    {
+        var filePath = Path.Combine(baseDir, Path.GetFileName(relativePath));
+        if (File.Exists(filePath))
+            return File.ReadAllText(filePath);
+
+        var baseName = Assembly.GetExecutingAssembly().GetName().Name;
+        var resourceName = relativePath
+            .TrimStart('.')
+            .Replace(Path.DirectorySeparatorChar, '.')
+            .Replace(Path.AltDirectorySeparatorChar, '.');
+
+        var manifestResourceName = Assembly.GetExecutingAssembly()
+            .GetManifestResourceNames().FirstOrDefault(x => x.EndsWith(resourceName));
+
+        if (string.IsNullOrEmpty(manifestResourceName))
+            throw new InvalidOperationException($"Did not find required resource ending in '{resourceName}' in assembly '{baseName}'.");
+
+        using var stream = Assembly.GetExecutingAssembly()
+            .GetManifestResourceStream(manifestResourceName);
+
+        if (stream == null)
+            throw new InvalidOperationException($"Did not find required resource '{manifestResourceName}' in assembly '{baseName}'.");
+
+        using var reader = new StreamReader(stream);
+        return reader.ReadToEnd();
+    }
+}

--- a/src/dotnet-nugetize/Program.cs
+++ b/src/dotnet-nugetize/Program.cs
@@ -42,6 +42,7 @@ namespace NuGetize
             .Add("  [msbuild args]             Examples: ")
             .Add("                             - Automatically restore: -r")
             .Add("                             - Set normal MSBuild verbosity: -v:n")
+            .Add("                             - Set minimal MSBuild verbosity: -v:m")
             .Add("                             - Build configuration: -p:Configuration=Release");
 
             extra = options.Parse(args);
@@ -116,18 +117,53 @@ namespace NuGetize
             if (File.Exists(file))
                 File.Delete(file);
 
-            // Optimize the "build" so that it doesn't actually do a full compile if possible.
+            var slnTargets = $"after.{Path.GetFileName(project)}.targets";
+            var deleteSlnTargets = false;
             var contentsOnly = false;
-            if (!Execute(DotnetMuxer.Path.FullName, $"msbuild {project} {string.Join(' ', extra)} -p:dotnet-nugetize=\"{file}\" -t:\"GetPackageContents;Pack\""))
-                return -1;
+
+            try
+            {
+                if (project.EndsWith(".sln") && !File.Exists(slnTargets))
+                {
+                    File.WriteAllText(slnTargets, EmbeddedResource.GetContent("after.sln.targets"), Encoding.UTF8);
+                    deleteSlnTargets = true;
+                }
+
+                // Optimize the "build" so that it doesn't actually do a full compile if possible.
+                if (!Execute(DotnetMuxer.Path.FullName, $"msbuild {project} {string.Join(' ', extra)} -p:dotnet-nugetize=\"{file}\" -t:\"GetPackageContents;Pack\""))
+                {
+                    // The error might have been caused by us not being able to write our slnTargets
+                    if (project.EndsWith(".sln") && !deleteSlnTargets)
+                        ColorConsole.WriteLine($"Solution targets '{slnTargets}' already exists. NuGetizing all projects in the solution is therefore required.".Yellow());
+
+                    return -1;
+                }
+
+                if (!File.Exists(file))
+                {
+                    // Re-run requesting contents only.
+                    if (!Execute(DotnetMuxer.Path.FullName, $"msbuild {project} {string.Join(' ', extra)} -p:dotnet-nugetize-contents=true -p:dotnet-nugetize=\"{file}\" -t:\"GetPackageContents\""))
+                    {
+                        // The error might have been caused by us not being able to write our slnTargets
+                        if (project.EndsWith(".sln") && !deleteSlnTargets)
+                            ColorConsole.WriteLine($"Solution targets '{slnTargets}' already exists. NuGetizing all projects in the solution is therefore required.".Yellow());
+
+                        return -1;
+                    }
+
+                    contentsOnly = true;
+                }
+            }
+            finally
+            {
+                if (File.Exists(slnTargets) && deleteSlnTargets)
+                    File.Delete(slnTargets);
+            }
 
             if (!File.Exists(file))
             {
-                // Re-run requesting contents only.
-                if (!Execute(DotnetMuxer.Path.FullName, $"msbuild {project} {string.Join(' ', extra)} -p:dotnet-nugetize-contents=true -p:dotnet-nugetize=\"{file}\" -t:\"GetPackageContents\""))
-                    return -1;
-
-                contentsOnly = true;
+                ColorConsole.WriteLine("Failed to discover nugetized content.".Red());
+                return -1;
             }
 
             var doc = XDocument.Load(file);

--- a/src/dotnet-nugetize/after.sln.targets
+++ b/src/dotnet-nugetize/after.sln.targets
@@ -1,0 +1,17 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="Current" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Target Name="RemoveNonNuGetized" BeforeTargets="GetPackageContents">
+    <MSBuild Properties="BuildingSolutionFile=true; CurrentSolutionConfigurationContents=$(CurrentSolutionConfigurationContents); SolutionDir=$(SolutionDir); SolutionExt=$(SolutionExt); SolutionFileName=$(SolutionFileName); SolutionName=$(SolutionName); SolutionPath=$(SolutionPath)" 
+             BuildInParallel="True" 
+             SkipNonexistentProjects="%(ProjectReference.SkipNonexistentProjects)" 
+             Targets="GetTargetPath" 
+             Projects="@(ProjectReference)">
+      <Output TaskParameter="TargetOutputs" ItemName="ReferencedProjectTargetPath" />
+    </MSBuild>
+    <ItemGroup>
+      <NuGetizedTargetPath Include="@(ReferencedProjectTargetPath -> WithMetadataValue('IsNuGetized', 'true'))" />
+      <NonNuGetizedTargetPath Include="@(ReferencedProjectTargetPath)" Exclude="@(NuGetizedTargetPath)" />
+      <ProjectReference Remove="@(NonNuGetizedTargetPath -> '%(MSBuildSourceProjectFile)')" />
+    </ItemGroup>
+  </Target>
+</Project>

--- a/src/dotnet-nugetize/dotnet-nugetize.csproj
+++ b/src/dotnet-nugetize/dotnet-nugetize.csproj
@@ -29,6 +29,8 @@
   <ItemGroup>
     <None Update="NuGetize.Build.targets" CopyToOutputDirectory="PreserveNewest" />
     <Compile Remove="Range.cs" Condition="'$(TargetFramework)' != 'netcoreapp2.1'" />
+    <None Remove="after.sln.targets" />
+    <EmbeddedResource Include="after.sln.targets" CopyToOutputDirectory="PreserveNewest" />
     <ProjectProperty Include="Version" />
     <ProjectProperty Include="ToolCommandName" />
     <ProjectProperty Include="Product" />    

